### DISCUSSION TITLE
C Device Service - Retrieve Query String From GET Request

### DISCUSSION THREAD

**Discussion by vijay-hcl-epl** [edgexfoundry/community#204] [Q&A] [answered]
Hi Team,
I am developing a custom C device-service in which I need to receive Query string from the GET request in the get_handler. As per the [C servicewritersguide](https://github.com/edgexfoundry/device-sdk-c/blob/main/docs/servicewritersguide.md) , in get_handler, iot_data_t *options can be used to get the query string from the GET request URL.

**Queries**
1. Is there any reference code available which gets the query string from the options parameter in get_handler?
2. I have used iot_data_to_json API to convert option to json. Now I have the query string as json key-value pair. Is there any in-built json API available in device-sdk-c/EdgeX which I can use to retrieve the value from the key-value pair? If it is not available I am planning to use external libraries such as json-c or jansson C library. 

Please give your comments.

Thank you,
Vijay

**Comment by iain-anderson** [comment, marked as answer]
Hi Vijay, I don't know of anyone who has used this yet. However, `options` is a string map; you can use `iot_data_string_map_get_string(options, "MY_KEY")` to retrieve a value

**Reply by vijay-hcl-epl** [reply]
Thank you @iain-anderson for the information. It helps..!!
